app/src/main/java/app/revanced/manager/ui/component/InstallerStatusDialog.kt

```diff
@@ -0,0 +1,166 @@
+    val confirmButton: InstallerStatusDialogButton = installerStatusDialogButton(R.string.ok),
+    val dismissButton: InstallerStatusDialogButton? = null,
+) {
+    FAILURE(
+        flag = PackageInstaller.STATUS_FAILURE,
+        title = R.string.installation_failed_dialog_title,
+        contentStringResId = R.string.installation_failed_description,
+        confirmButton = installerStatusDialogButton(R.string.install_app) { model ->
+            model.install()
+        }
+    ),
+    FAILURE_ABORTED(
+        flag = PackageInstaller.STATUS_FAILURE_ABORTED,
+        title = R.string.installation_cancelled_dialog_title,
+        contentStringResId = R.string.installation_aborted_description,
+        confirmButton = installerStatusDialogButton(R.string.install_app) { model ->
+            model.install()
+        }
+    ),
+    FAILURE_BLOCKED(
+        flag = PackageInstaller.STATUS_FAILURE_BLOCKED,
+        title = R.string.installation_blocked_dialog_title,
+        contentStringResId = R.string.installation_blocked_description,
+    ),
+    FAILURE_CONFLICT(
+        flag = PackageInstaller.STATUS_FAILURE_CONFLICT,
+        title = R.string.installation_conflict_dialog_title,
+        contentStringResId = R.string.installation_conflict_description,
+        confirmButton = installerStatusDialogButton(R.string.reinstall) { model ->
+            model.reinstall()
+        },
+        dismissButton = installerStatusDialogButton(R.string.cancel),
+    ),
+    FAILURE_INCOMPATIBLE(
+        flag = PackageInstaller.STATUS_FAILURE_INCOMPATIBLE,
+        title = R.string.installation_incompatible_dialog_title,
+        contentStringResId = R.string.installation_incompatible_description,
+    ),
+    FAILURE_INVALID(
+        flag = PackageInstaller.STATUS_FAILURE_INVALID,
+        title = R.string.installation_invalid_dialog_title,
+        contentStringResId = R.string.installation_invalid_description,
+        confirmButton = installerStatusDialogButton(R.string.reinstall) { model ->
+            model.reinstall()
+        },
+        dismissButton = installerStatusDialogButton(R.string.cancel),
+    ),
+    FAILURE_STORAGE(
+        flag = PackageInstaller.STATUS_FAILURE_STORAGE,
+        title = R.string.installation_storage_issue_dialog_title,
+        contentStringResId = R.string.installation_storage_issue_description,
+    ),
+
+    @RequiresApi(34)
+    FAILURE_TIMEOUT(
+        flag = PackageInstaller.STATUS_FAILURE_TIMEOUT,
+        title = R.string.installation_timeout_dialog_title,
+        contentStringResId = R.string.installation_timeout_description,
+        confirmButton = installerStatusDialogButton(R.string.install_app) { model ->
+            model.install()
+        },
+    ),
+    SUCCESS(
+        flag = PackageInstaller.STATUS_SUCCESS,
+        title = R.string.installation_success_dialog_title,
+        contentStringResId = R.string.installation_success_description,
+        icon = Icons.Outlined.Check,
+    );
+
```

Comment:
```suggestion
```

Because of https://github.com/ReVanced/revanced-manager/pull/1473#issuecomment-2254259701